vars/database.yml

```diff
@@ -0,0 +1,50 @@
+---
+database_mode: internal
+database_host: localhost
+database_port: 5432
+database_ssl_mode: disable
+database_ssl_ca: None
+
+foreman_database_name: foreman
+foreman_database_user: foreman
+foreman_database_password: CHANGEME
+candlepin_database_name: candlepin
+candlepin_database_user: candlepin
+candlepin_database_password: CHANGEME
+pulp_database_name: pulp
+pulp_database_user: pulp
+pulp_database_password: CHANGEME
+
+candlepin_database_host: "{{ database_host }}"
+candlepin_database_port: "{{ database_port }}"
+candlepin_database_ssl_mode: "{{ database_ssl_mode }}"
+candlepin_database_ssl_ca: "{{ database_ssl_ca }}"
+
+pulp_settings_database_env:
+  PULP_DATABASES__default__NAME: "{{ pulp_database_name }}"
+  PULP_DATABASES__default__USER: "{{ pulp_database_user }}"
+  PULP_DATABASES__default__PASSWORD: "{{ pulp_database_password }}"
```

Comment:
this needs to be a secret, the others can remain in the main "env dict"

Reply:
And because that's already done in `roles/pulp/tasks/main.yaml`, you can just
```suggestion
```

Reply:
but at the same time, I don't think you need to define `pulp_settings_database_env` here -- `pulp_database_*` is the input to the `pulp` role, which is all we should care about in the vars file.

Reply:
I don't see for example `pulp_database_host` in the Pulp role anywhere. What am I missing about the design?

Reply:
Should I be adding podman secret blocks for each? Or should I be adding these to defaults? Or changing https://github.com/theforeman/foreman-quadlet/blob/master/roles/pulp/defaults/main.yaml#L21 to not be hardcoded to `localhost` but instead to `pulp_database_host` ?

Reply:
Well, it's not yet defined (as we only needed localhost), but the structure is here:
https://github.com/theforeman/foreman-quadlet/blob/a203f0170a23e5159028855bd8ffb3ab7211e2c7/roles/pulp/defaults/main.yaml#L20-L24

so you could change that to
```yaml
pulp_database_name: pulp
pulp_database_user: pulp
pulp_database_host: localhost
pulp_database_port: 5432

pulp_settings_db_env:
  PULP_DATABASES__default__HOST: "{{ pulp_database_host }}"
  PULP_DATABASES__default__NAME: "{{ pulp_database_name }}"
  PULP_DATABASES__default__PORT: "{{ pulp_database_port }}"
  PULP_DATABASES__default__USER: "{{ pulp_database_user }}"
```

and then this database.yml only needs to override `pulp_database_*`